<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="The goal of `tpSVG` is to detect and visualize spatial variation in
  the gene expression for spatially resolved transcriptomics data analysis.
  Specifically, `tpSVG` introduces a family of count-based models, with 
  generalizable parametric assumptions such as Poisson distribution or 
  negative binomial distribution. In addition, comparing to currently 
  available count-based model for spatially resolved data analysis, the 
  `tpSVG` models improves computational time, and hence greatly improves 
  the applicability of count-based models in SRT data analysis. ">
<title>Thin plate models to detect spatially variable genes  • tpSVG</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="apple-touch-icon-60x60.png">
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Thin plate models to detect spatially variable genes ">
<meta property="og:description" content="The goal of `tpSVG` is to detect and visualize spatial variation in
  the gene expression for spatially resolved transcriptomics data analysis.
  Specifically, `tpSVG` introduces a family of count-based models, with 
  generalizable parametric assumptions such as Poisson distribution or 
  negative binomial distribution. In addition, comparing to currently 
  available count-based model for spatially resolved data analysis, the 
  `tpSVG` models improves computational time, and hence greatly improves 
  the applicability of count-based models in SRT data analysis. ">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">tpSVG</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.7</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="articles/intro_to_tpSVG.html">Introduction to `tpSVG`</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/boyiguo1/tpSVG/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header">
<img src="logo.png" class="logo" alt=""><h1 id="tpsvg-">tpSVG <a class="anchor" aria-label="anchor" href="#tpsvg-"></a>
</h1>
</div>
<!-- badges: start -->

<p>The goal of <code>tpSVG</code> is to detect and visualize spatial variation in the gene expression for spatially resolved transcriptomics data analysis. Specifically, <code>tpSVG</code> introduces a family of count-based models, with generalizable parametric assumptions such as Poisson distribution or negative binomial distribution. In addition, comparing to crmarkdown::pandoc_version()urrently available count-based model for spatially resolved data analysis, the <code>tpSVG</code> models improves computational time, and hence greatly improves the applicability of count-based models in SRT data analysis.</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="section level3">
<h3 id="github">GitHub<a class="anchor" aria-label="anchor" href="#github"></a>
</h3>
<p>You can install the development version of tpSVG from <a href="https://github.com/boyiguo1/tpSVG" class="external-link">GitHub</a> with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#' Install devtools package if not already installed</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu">required</span><span class="op">(</span><span class="va">devtools</span><span class="op">)</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="va">package_name</span><span class="op">)</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"boyiguo1/tpSVG"</span><span class="op">)</span></span></code></pre></div>
<p>If you have R version before v4.4 and would like to install tpSVG, you can follow</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">"devtools"</span>)) <span class="fu">install.packages</span>(<span class="st">"devtools"</span>)</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">"boyiguo1/tpSVG@pre-R4.4"</span>)</span></code></pre></div>
<blockquote>
<p>WARNING: The purpose of having the branch pre-R4.4 is to allow users to use escheR before the formal release of R 4.4 and during the early stage of R 4.4 release. This branch will not be update with any further development beyond escheR v0.99.1. We recommend users to update their R versions up to date.</p>
</blockquote>
</div>
<div class="section level3">
<h3 id="bioconductor-pending">Bioconductor (pending)<a class="anchor" aria-label="anchor" href="#bioconductor-pending"></a>
</h3>
<p>The package is currently submitted to Bioconductor for <a href="https://github.com/Bioconductor/Contributions/issues/3264" class="external-link">review</a>. Once the package is accepted by Bioconductor, you can install the latest release version of <code>tpSVG</code> from Bioconductor via the following code. Additional details are shown on the Bioconductor page.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># NOTE: The package is under-review with bioconductor.</span></span>
<span><span class="co">#       The following code section will work once the package is accepted.</span></span>
<span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">"BiocManager"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"tpSVG"</span><span class="op">)</span></span></code></pre></div>
<p>The latest development version can also be installed from the <code>devel</code> version of Bioconductor or from GitHub following</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span>version <span class="op">=</span> <span class="st">"devel"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="tutorial">Tutorial<a class="anchor" aria-label="anchor" href="#tutorial"></a>
</h2>
<p>Please find an end-to-end tutorial at <a href="https://boyi-guo.com/tpSVG/articles/intro_to_tpSVG.html" class="external-link uri">https://boyi-guo.com/tpSVG/articles/intro_to_tpSVG.html</a>.</p>
</div>
<div class="section level2">
<h2 id="frequently-asked-questions">Frequently asked questions<a class="anchor" aria-label="anchor" href="#frequently-asked-questions"></a>
</h2>
<p><strong>Implementation Questions</strong></p>
<ul>
<li>
<p>What are the data structures that <code>tpSVG</code> current supports?</p>
<p><em>As of <code>tpSVG v0.99.1</code>, the data structure <code>tpSVG</code> supports includes <a href="https://bioconductor.org/packages/release/bioc/html/SpatialFeatureExperiment.html" class="external-link"><code>SpatialExperiments</code></a> (and packages extending <code>SpatialExperiments</code>, e.g. <a href="https://bioconductor.org/packages/release/bioc/html/SpatialFeatureExperiment.html" class="external-link"><code>SpatialFeatureExperiments</code></a>) and <code>data.frame</code>. Please find example via <a href="https://boyi-guo.com/escheR/articles/supported_data_structure.html" class="external-link">supported_data_structure</a>. Due to limited resources, we regret that we won’t provides direct accessibility to other pipelines, e.g. <code>suerat</code>.</em></p>
</li>
<li>
<p>What types of spatially-resolved transcriptomics (SRT) data that <code>tpSVG</code> supports?</p>
<p><em>Both sequenced-based SRT and image-based SRT data are supported by <code>tpSVG</code>. For more details, please refer to the vignette <a href="https://boyi-guo.com/tpSVG/articles/supported_data_structure.html#image-based-srt-in-spatialexperiment-e-g--spatialfeatureexperiment" class="external-link">supported_data_structure</a>.</em></p>
</li>
<li>
<p>Can I use other scale factor as offset in the count-model?</p>
<p><em>Yes, just remember to take log for the offset term. In the vignettes, the offset of the model is default to library size, i.e. the total number of molecular in a spot/cell, but the count models should be compatible to other definition of scale factor in theory.</em></p>
</li>
</ul>
<p><strong>Theoretical Questions</strong></p>
<ul>
<li>
<p>What is the difference between modeling log transformed data and count data?</p>
<p><em>Count data is the natural form of gene expression data when it is collected and quantified. While log-transformation providess shortcuts to model (normalized) count data using well-studied Gaussian distribution, it distorts the lowly expressed gene and causes analytic biases.</em></p>
</li>
</ul>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/boyiguo1/tpSVG/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/boyiguo1/tpSVG/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing tpSVG</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Boyi Guo <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0003-2950-2349" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Stephanie C. Hicks <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-7858-0231" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li><a href="authors.html">More about authors...</a></li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental" class="external-link"><img src="https://img.shields.io/badge/lifecycle-experimental-orange.svg" alt="Lifecycle: experimental"></a></li>
<li><a href="https://github.com/boyiguo1/tpSVG/actions" class="external-link"><img src="https://github.com/boyiguo1/tpSVG/workflows/R-CMD-check-bioc/badge.svg" alt="R build status"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Boyi Guo, Stephanie C. Hicks.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
